File: sgi-frontend/src/types/seg/rol.ts
import { Permiso } from './permiso';

export interface RolBase {
    nombre: string;
    descripcion: string | null;
    is_active: boolean;
}

export interface RolCreate extends RolBase {
    permisos_ids: number[];
}

export interface RolUpdate {
    nombre?: string | null;
    descripcion?: string | null;
    is_active?: boolean | null;
    permisos_ids?: number[] | null;
}

export interface Rol extends RolBase {
    id: number;
    permisos: Permiso[];
}

File: sgi-backend/app/schemas/seg/rol.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List
from app.schemas.seg.permiso import PermisoResponse

# =========================================
# SCHEMAS DE ROLES
# =========================================

class RolBase(BaseModel):
    nombre: str
    descripcion: Optional[str] = None
    is_active: bool = True

class RolCreate(RolBase):
    permisos_ids: List[int] = []

class RolUpdate(BaseModel):
    nombre: Optional[str] = None
    descripcion: Optional[str] = None
    is_active: Optional[bool] = None
    permisos_ids: Optional[List[int]] = None

class RolResponse(RolBase):
    id: int
    permisos: List[PermisoResponse] = []

    model_config = ConfigDict(from_attributes=True)
